core/modules/replica/Configuration.cc

```diff
@@ -0,0 +1,333 @@
+std::string  const Configuration::defaultDatabasePassword            ("");
+std::string  const Configuration::defaultDatabaseName                ("replica");
+unsigned int const Configuration::defaultJobSchedulerIvalSec         (1);
+size_t       const Configuration::defaultReplicationLevel            (1);
+
+void Configuration::translateDataDir(std::string&       dataDir,
+                                     std::string const& workerName) {
+
+    std::string::size_type const leftPos = dataDir.find('{');
+    if (leftPos == std::string::npos) { return; }
+
+    std::string::size_type const rightPos = dataDir.find('}');
+    if (rightPos == std::string::npos) { return; }
+
+    if (rightPos <= leftPos) {
+        throw std::invalid_argument(
+                "Configuration::translateDataDir  misformed template in the data directory path: '" +
+                dataDir + "'");
+    }
+    if (dataDir.substr (leftPos, rightPos - leftPos + 1) == "{worker}") {
+        dataDir.replace(leftPos, rightPos - leftPos + 1, workerName);
+    }
+}
+
+Configuration::Configuration()
+    :   _requestBufferSizeBytes       (defaultRequestBufferSizeBytes),
+        _retryTimeoutSec              (defaultRetryTimeoutSec),
+        _controllerThreads            (defaultControllerThreads),
+        _controllerHttpPort           (defaultControllerHttpPort),
+        _controllerHttpThreads        (defaultControllerHttpThreads),
+        _controllerRequestTimeoutSec  (defaultControllerRequestTimeoutSec),
+        _jobTimeoutSec                (defaultJobTimeoutSec),
+        _jobHeartbeatTimeoutSec       (defaultJobHeartbeatTimeoutSec),
+        _xrootdAutoNotify             (defaultXrootdAutoNotify),
+        _xrootdHost                   (defaultXrootdHost),
+        _xrootdPort                   (defaultXrootdPort),
+        _xrootdTimeoutSec             (defaultXrootdTimeoutSec),
+        _workerTechnology             (defaultWorkerTechnology),
+        _workerNumProcessingThreads   (defaultWorkerNumProcessingThreads),
+        _fsNumProcessingThreads       (defaultFsNumProcessingThreads),
+        _workerFsBufferSizeBytes      (defaultWorkerFsBufferSizeBytes),
+        _databaseTechnology           (defaultDatabaseTechnology),
+        _databaseHost                 (defaultDatabaseHost),
+        _databasePort                 (defaultDatabasePort),
+        _databaseUser                 (defaultDatabaseUser),
+        _databasePassword             (defaultDatabasePassword),
+        _databaseName                 (defaultDatabaseName),
+        _jobSchedulerIvalSec          (defaultJobSchedulerIvalSec) {
+}
+
+std::string Configuration::context() const {
+    static std::string const str = "CONFIG   ";
+    return str;
+}
+
+std::vector<std::string> Configuration::workers(bool isEnabled,
+                                                bool isReadOnly) const {
+    std::vector<std::string> names;
+    for (auto&& entry: _workerInfo) {
```

Comment:
this is happening in a single threaded environment?

Reply:
or, the map is unchanged after init time?

Reply:
well, no: `ConfigurationFile::deleteWorker`

Reply:
Good point. Sealed to work in a multithread environment. Though, this area is still under development. Certain aspects (such as reconfiguring the application at run-time w/o restarting it) will be eventually refined to work in k8 deployments (once we'll have a definitive plan for those deployments).

Reply:
by "sealed" do you mean the map is protected against concurrent access? If so, it would be good to at least add comments or the pass-in lock guard pattern, as we've discussed in other contexts.

Reply:
Each of those methods now gins with:
util::Lock lock(_mtx, context() + "workers");

GitHub still shows an outdated version of the code.